src/args.rs

```diff
@@ -57,6 +59,40 @@ impl From<DeviceModel> for nitrokey::Model {
   }
 }
 
+impl<'de> serde::Deserialize<'de> for DeviceModel {
```

Comment:
Exactly.  If we move forward with this approach, I’m going to fix the macro so that we don’t need this implementation.

Reply:
Sounds good

Reply:
Fixing the macro will be a bit harder than I remembered because it already defines some derives.  As a first step, I simplified the manual Deserialize implementation.

Reply:
I pretty much thought so. Yeah, that looks fine to me.